pkg/reconciler/configuration/config/store.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+
+	"knative.dev/pkg/configmap"
+	cfgmap "knative.dev/serving/pkg/apis/config"
+)
+
+type cfgKey struct{}
+
+// Config holds the collection of configurations that we attach to contexts.
+// +k8s:deepcopy-gen=false
+type Config struct {
+	Defaults *cfgmap.Defaults
+	Features *cfgmap.Features
+}
+
+// FromContext extracts a Config from the provided context.
+func FromContext(ctx context.Context) *Config {
+	x, ok := ctx.Value(cfgKey{}).(*Config)
+	if ok {
+		return x
+	}
+	return nil
+}
+
+// FromContextOrDefaults is like FromContext, but when no Config is attached it
+// returns a Config populated with the defaults for each of the Config fields.
+func FromContextOrDefaults(ctx context.Context) *Config {
+	cfg := FromContext(ctx)
+	if cfg == nil {
+		cfg = &Config{}
+	}
+
+	if cfg.Defaults == nil {
+		cfg.Defaults, _ = cfgmap.NewDefaultsConfigFromMap(map[string]string{})
```

Comment:
I think `nil` map here and below should work just fine.

Reply:
actually not because the responsive revision depends on features

Reply:
Sure. I don't see how that relates to my comment though? :)

Reply:
defaults can be nil but features cannot as  it gets referenced [in responsive revision](https://github.com/knative/serving/blob/6826f9a6f80be394552e81908edb4a4086c56d14/pkg/reconciler/configuration/resources/revision.go#L46)  (we use the same store there)

Reply:
You're reading the comment incorrectly. I am not saying that Defaults CM or Features CM can be nil, I mean the map you're passing to the call. You create unnecessary objects.

Reply:
Done!